src/interface/sparse_pack.hpp

```diff
@@ -161,6 +161,13 @@ struct any : public base_t<true> {
   KOKKOS_INLINE_FUNCTION any(Ts &&...args) : base_t<true>(std::forward<Ts>(args)...) {}
   static std::string name() { return ".*"; }
 };
+
+struct any_nonflux : public base_t<true> {
+  template <class... Ts>
+  KOKKOS_INLINE_FUNCTION any_nonflux(Ts &&...args)
+      : base_t<true>(std::forward<Ts>(args)...) {}
+  static std::string name() { return "^(?!bnd_flux::).+"; }
```

Comment:
Should it be documented somewhere that fluxes have this name? Actually for output purposes we probably want that too. i.e., we want to be able to output fluxes for debugging. So users should know what the name format is. And also should be forbidden from trying to manually register variables with this pattern.

Reply:
yeah, maybe we should disallow `::` in variable names? That would give us some flexibility for creating more auto-generated variables. Also `bnd_flux` was just what Josh had chosen in the his old PR making fluxes variables. We could change that,

Also, I guess there is no rule that fluxes have to be named `bnd_flux::*`. A downstream code could declare a field with `Metadata::Flux` with any name, not explicitly associate it with any other field, and it would still get communicated during the flux correction routines. Maybe this should read

Reply:
I am in favor of disallowing `::` in variable names. And agree about `any_nonautoflux`

Reply:
renamed `any_nonflux` but keeping this conversation open to hear others thoughts on reserving `::`

Reply:
The `::` may or may not cause a problem as it's adisallowed character by the OpenPMD standard.
I'm actually not sure if this is a hard problem or not (as I assume that some default tools may not work with that approach, but we probably would dump these flux variable only for debugging anyway, don't we)?
Also, we could in principle overwrite the reserved pattern for output and do a name matching when reading them back in again.

Reply:
Checking that people don't add variables with `::` is considerably harder because of how all the package resolution works. I went down that path for a little while and eventually gave up.